src/Datadog.AutoInstrumentation.NativeLoader/cor_profiler.cpp

```diff
@@ -0,0 +1,4587 @@
+                }
+                else
+                {
+                    Warn("Error getting the event mask.");
+                }
+            }
+            else
+            {
+                Warn("Error Initializing the Continuous Profiler.");
+                return localResult;
+            }
+        }
+
+        //
+        // Tracer Initialization
+        //
+        if (m_tracerProfiler != nullptr)
+        {
+            HRESULT localResult = m_tracerProfiler->Initialize(pICorProfilerInfoUnk);
+            if (SUCCEEDED(localResult))
+            {
+                // let's get the event mask set by the CP.
+                DWORD local_mask_low;
+                DWORD local_mask_hi;
+                HRESULT hr = info6->GetEventMask2(&local_mask_low, &local_mask_hi);
+                if (SUCCEEDED(hr))
+                {
+                    mask_low = mask_low | local_mask_low;
+                    mask_hi = mask_hi | local_mask_hi;
+
+                    Debug("*LocalMaskLow: ", local_mask_low);
+                    Debug("*LocalMaskHi : ", local_mask_hi);
+                }
+                else
+                {
+                    Warn("Error getting the event mask.");
+                }
+            }
+            else
+            {
+                Warn("Error Initializing the Tracer Profiler.");
+                return localResult;
+            }
+        }
+
+        //
+        // Custom Profiler Initialization
+        //
+        if (m_customProfiler != nullptr)
+        {
+            HRESULT localResult = m_customProfiler->Initialize(pICorProfilerInfoUnk);
+            if (SUCCEEDED(localResult))
+            {
+                // let's get the event mask set by the CP.
+                DWORD local_mask_low;
+                DWORD local_mask_hi;
+                HRESULT hr = info6->GetEventMask2(&local_mask_low, &local_mask_hi);
+                if (SUCCEEDED(hr))
+                {
+                    mask_low = mask_low | local_mask_low;
+                    mask_hi = mask_hi | local_mask_hi;
+
+                    Debug("*LocalMaskLow: ", local_mask_low);
+                    Debug("*LocalMaskHi : ", local_mask_hi);
+                }
+                else
+                {
+                    Warn("Error getting the event mask.");
+                }
+            }
+            else
+            {
+                Warn("Error Initializing the Custom Profiler.");
+                return localResult;
+            }
+        }
+
+        //
+        // Sets final event mask as a combination of each cor profiler masks.
+        //
+        Debug("*MaskLow: ", mask_low);
+        Debug("*MaskHi : ", mask_hi);
+
+        // Sets the final event mask for the profiler
+        hr = info6->SetEventMask2(mask_low, mask_hi);
+        if (FAILED(hr))
+        {
+            Warn("Error setting the event mask.");
+            return E_FAIL;
+        }
+
+        return S_OK;
```

Comment:
We are already logging on each local profiler FAILED condition. About if we should return OK or not in case at least one is OK we should talk about it with @gleocadie , I don't have any strong opinion but is a behaviour change we must agree.

Reply:
Logs added in 4d06dd5

Reply:
✔️